src/client/config.rs

```diff
@@ -21,7 +38,9 @@ impl PlatzClientConfig {
     /// new_from_secret, whichever succeeds first.
     /// If no token can be found, a PlatzClientError::NotFound is returned.
     pub async fn new() -> Result<Self, PlatzClientError> {
-        if let Some(config) = Self::new_from_env()? {
+        if let Some(config) = Self::new_from_configuration(None).await? {
```

Comment:
Environment variables usually precede configuration files, I think the order should be env, then config files, and only then from secret.